pkg/gateway/main_test.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/treeverse/lakefs/pkg/gateway/multiparts"
+	"github.com/treeverse/lakefs/pkg/gateway/simulator"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/kv/kvtest"
+	_ "github.com/treeverse/lakefs/pkg/kv/mem"
+	"github.com/treeverse/lakefs/pkg/stats"
+	"github.com/treeverse/lakefs/pkg/testutil"
+)
+
+type dependencies struct {
+	blocks  block.Adapter
+	auth    simulator.GatewayAuthService
+	catalog catalog.Interface
+}
+
+const ReplayRepositoryName = "example"
+
+var (
+	pool         *dockertest.Pool
+	databaseURI  string
+	IdTranslator *testutil.UploadIDTranslator
+)
+
+type mockCollector struct{}
+
+func (m *mockCollector) CollectMetadata(*stats.Metadata) {}
+
+func (m *mockCollector) CollectEvent(string, string) {}
+
+func (m *mockCollector) SetInstallationID(string) {}
+
+func (m *mockCollector) Close() {}
+
+func getBasicHandler(t *testing.T, authService *simulator.PlayBackMockConf, kvEnabled bool) (http.Handler, *dependencies) {
+	ctx := context.Background()
+	IdTranslator = &testutil.UploadIDTranslator{TransMap: make(map[string]string),
+		ExpectedID: "",
+		T:          t,
+	}
+	viper.Set(config.BlockstoreTypeKey, block.BlockstoreTypeMem)
+	conf, err := config.NewConfig()
+	testutil.MustDo(t, "config", err)
+	// Do not validate invalid config (missing required fields).
+
+	conn, _ := testutil.GetDB(t, databaseURI)
+	c, err := catalog.New(ctx, catalog.Config{
+		Config: conf,
+		DB:     conn,
+	})
+	testutil.MustDo(t, "build c", err)
+
+	var multipartsTracker multiparts.Tracker
+	if kvEnabled {
+		store := kvtest.MakeStoreByName("mem", "")(t, context.Background())
+		defer store.Close()
+		multipartsTracker = multiparts.NewTracker(kv.StoreMessage{Store: store})
+	} else {
+		multipartsTracker = multiparts.NewDBTracker(conn)
+	}
+
+	blockstoreType, _ := os.LookupEnv(testutil.EnvKeyUseBlockAdapter)
+	blockAdapter := testutil.NewBlockAdapterByType(t, IdTranslator, blockstoreType)
+
+	t.Cleanup(func() {
+		_ = c.Close()
+	})
+
+	storageNamespace := os.Getenv("USE_STORAGE_NAMESPACE")
+	if storageNamespace == "" {
+		storageNamespace = "replay"
+	}
+
+	// keeping the name 'master' and not 'main' below as the recordings point to that
+	_, err = c.CreateRepository(ctx, ReplayRepositoryName, storageNamespace, "master")
+	testutil.Must(t, err)
+
+	handler := gateway.NewHandler(authService.Region, c, multipartsTracker, blockAdapter, authService, []string{authService.BareDomain}, &mockCollector{}, nil, false)
+
+	return handler, &dependencies{
+		blocks:  blockAdapter,
+		auth:    authService,
+		catalog: c,
+	}
+}
+
+func TestMain(m *testing.M) {
+	var err error
+	var closer func()
+	pool, err = dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to Docker: %s", err)
+	}
+	databaseURI, closer = testutil.GetDBInstance(pool)
+	code := m.Run()
+	closer() // cleanup
+	os.Exit(code)
+}
+
+func setupTest(t *testing.T, method, target string, body io.Reader, kvEnabled bool) *http.Response {
```

Comment:
I'd rather not put a lot of thought into it as this is temporary code just until we complete the feature

Reply:
This code won't be removed (checking with @guy-har if the test using it can be removed altogether) but the kvEnabled flag will no longer be relevant once we are in feature complete and we will only be using and testing one tracker implementation